fendermint/vm/interpreter/src/fvm/query.rs

```diff
@@ -155,6 +164,16 @@ where
                 let (state, ret) = state.builtin_actors().await?;
                 Ok((state, FvmQueryRet::BuiltinActors(ret)))
             }
+            FvmQuery::UpgradeSchedule => {
+                let ret = self.upgrade_schedule.get_all();
+                Ok((state, FvmQueryRet::UpgradeSchedule(ret)))
+            }
+            FvmQuery::NodeState => {
+                let ret = NodeState {
+                    frozen: IS_FROZEN.load(Ordering::Relaxed),
```

Comment:
So this one is executed by the same interpreter, it can access it as a member variable.

Reply:
Maybe it could tell on which upgrade it has been frozen, instead of just a bool.